tests/test_progress.py

```diff
@@ -0,0 +1,205 @@
+import time
```

Comment:
I personally like to have the tests inside the module under test: it is sometimes required by distribution packagers to be able to run the tests (and as such it's better if it's part of the distributed files)